tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php

```diff
@@ -166,6 +166,17 @@ public function testDontFinishNoCache()
         self::assertCount(2, $this->sqlLogger->queries);
     }
 
+    public function testFetchAllAndFinishSavesCache()
+    {
+        $layerCache = new ArrayCache();
+        $stmt       = $this->connection->executeQuery('SELECT * FROM caching WHERE test_int > 500', [], [], new QueryCacheProfile(10, 'testcachekey', $layerCache));
+        $stmt->fetchAll();
+        $stmt->closeCursor();
+
+        $cachedData = $layerCache->fetch('testcachekey');
+        self::assertInternalType('array', $cachedData);
```

Comment:
> This is redundant. The assertion below is enough.

Sorry, I got confused. `self::assertCount(1, $cachedData)` verifies that the `$cachedData` is an array and contains exactly one element (I assume it's `['testcachekey' => /* ... */ ]`). Would it make more sense to replace it with `assertEquals(['testcachekey' => /* ... */ ], $cachedData)`.

Reply:
I'd rather to use AssertCount than AssertEquals, because testing specific expected array data makes a test weaker. As an example in this test case, the expected $cachedData from fetch would be the following:

`$expectedCacheData = ['query=SELECT * FROM caching WHERE test_int > 500&params=a:0:{}&types=a:0:{}&connectionParams=369e8eb1793ec435a92daabfe65369b94fd4317397cf48afbdd411a067779bd7' => []];`

As I've checked in the other test no one is using AsserEquals but AssertCount.